cli/bin/build-go.sh

```diff
@@ -57,3 +57,5 @@ case "$OSTYPE" in
     linux*) strip "${EXE_NAME}${SUFFIX}"
 esac
 print_file "${EXE_NAME}${SUFFIX}"
+
+go test
```

Comment:
@skonto Could you add this earlier, like around L. 37? So the logic would be: if the unit tests do not pass, we do not bother building the (Windows, OSX, Linux) executables.

Reply:
Ok I thought you build first and then test, sure.

Reply:
Done.

Reply:
@susanxhuynh I dont need rebasing I think I touched different files.

Reply:
@skonto My recent fix is needed to get one of the CI tests to pass (something changed in hdfs and I had to add a workaround). So, I think you have to include that fix in order for the Jenkins tests to pass for this PR.